File: 11-container-with-most-water/container-with-most-water.cs
public class Solution {
    public int MaxArea(int[] height) {
        int n = height.Length;
        int maxarea = 0;
        int i = 0;
        int j = n-1;
        while(i<j){
            int curWidth = j-i;
            int curHeight = Math.Min(height[i],height[j]);
            maxarea = Math.Max(maxarea,curWidth*curHeight);
            if(height[i]<=height[j]){
                i++;
            }
            else{
                j--;
            }
        }
        return maxarea;
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        j = len(height)-1
        maxw = 0
        i = 0
        while i < j:
            if height[i] < height[j]:
                if maxw < height[i]*(j-i):
                    maxw = height[i]*(j-i)
            else:
                if maxw < height[j]*(j-i):
                    maxw = height[j]*(j-i)
            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        return maxw
        
        
        
        
        
            

           
        